internal/controller/secret/reconciler.go

```diff
@@ -57,6 +57,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 
 	if needed, etcd := isFinalizerNeeded(secret.Name, etcdList); needed {
+		if hasFinalizer(secret) {
+			return ctrl.Result{}, nil
+		}
```

Comment:
Thanks for fixing it.
Seems like we are not catching this case in our UT.

Reply:
This is an enhancement to avoid a no-op of trying to add finalizer for a resource that already has finalizer. It isn't a separate feature/functionality by itself, so there's no test for it. If you're suggesting having a unit test for `hasFinalizer()` function itself, then that's probably an overkill, since the function simply executes `return sets.NewString(secret.Finalizers...).Has(common.FinalizerName)`, which is too small and functionally basic to have a unit test for.